src/object-store/src/parser/parser.cpp

```diff
@@ -93,11 +95,11 @@ struct group_pred : if_must< one< '(' >, pad< pred, blank >, one< ')' > > {};
 struct true_pred : pegtl_istring_t("truepredicate") {};
 struct false_pred : pegtl_istring_t("falsepredicate") {};
 
-struct not_pre : seq< sor< one< '!' >, pegtl_istring_t("not") > > {};
+struct not_pre : seq< sor< one< '!' >, string_operator_t("not") > > {};
 struct atom_pred : seq< opt< not_pre >, pad< sor< group_pred, true_pred, false_pred, comparison_pred >, blank > > {};
 
-struct and_op : pad< sor< two< '&' >, pegtl_istring_t("and") >, blank > {};
-struct or_op : pad< sor< two< '|' >, pegtl_istring_t("or") >, blank > {};
+struct and_op : pad< sor< two< '&' >, string_operator_t("and") >, blank > {};
+struct or_op : pad< sor< two< '|' >, string_operator_t("or") >, blank > {};
```

Comment:
Is a similar change needed for `TRUE`, `FALSE`, `NULL`, `TRUEPREDICATE`, `FALSEPREDICATE`, `CONTAINS`, `BEGINSWITH`, and `ENDSWITH`?